pkg/mount/mount.go

```diff
@@ -120,13 +138,150 @@ func (m *mounter) getDevicePathBySerialID(volumeID string) (string, error) {
 			return source, nil
 		}
 		if !os.IsNotExist(err) {
+			fmt.Printf("Not found: %s\n", err.Error())
 			return "", err
 		}
 	}
 
 	return "", nil
 }
 
+func (m *mounter) getDevicePathForXenServer(volumeID string) (string, error) {
+	for i := 'b'; i <= 'z'; i++ {
+		devicePath := fmt.Sprintf("/dev/xvd%c", i)
+		fmt.Printf("Checking XenServer device path: %s\n", devicePath)
+
+		if _, err := os.Stat(devicePath); err == nil {
+			isBlock, err := m.IsBlockDevice(devicePath)
+			if err == nil && isBlock {
+				if m.verifyXenServerDevice(devicePath, volumeID) {
+					fmt.Printf("Found and verified XenServer device: %s\n", devicePath)
+					return devicePath, nil
+				}
+			}
+		}
+	}
+	return "", fmt.Errorf("device not found for volume %s", volumeID)
+}
+
+func (m *mounter) verifyXenServerDevice(devicePath string, volumeID string) bool {
+	size, err := m.GetBlockSizeBytes(devicePath)
+	if err != nil {
+		fmt.Printf("Failed to get device size: %v\n", err)
+		return false
+	}
+	fmt.Printf("Device size: %d bytes\n", size)
+
+	mounted, err := m.isDeviceMounted(devicePath)
+	if err != nil {
+		fmt.Printf("Failed to check if device is mounted: %v\n", err)
+		return false
+	}
+	if mounted {
+		fmt.Printf("Device is already mounted: %s\n", devicePath)
+		return false
+	}
+
+	props, err := m.getDeviceProperties(devicePath)
+	if err != nil {
+		fmt.Printf("Failed to get device properties: %v\n", err)
+		return false
+	}
+	fmt.Printf("Device properties: %v\n", props)
+
+	return true
+}
+
+func (m *mounter) getDevicePathForVMware(volumeID string) (string, error) {
+	// Loop through /dev/sdb to /dev/sdz (/dev/sda -> the root disk)
+	for i := 'b'; i <= 'z'; i++ {
+		devicePath := fmt.Sprintf("/dev/sd%c", i)
+		fmt.Printf("Checking VMware device path: %s\n", devicePath)
+
+		if _, err := os.Stat(devicePath); err == nil {
+			isBlock, err := m.IsBlockDevice(devicePath)
+			if err == nil && isBlock {
+				// Use the same verification as for XenServer
+				if m.verifyVMwareDevice(devicePath, volumeID) {
+					fmt.Printf("Found and verified VMware device: %s\n", devicePath)
+					return devicePath, nil
+				}
+			}
+		}
+	}
+	return "", fmt.Errorf("device not found for volume %s", volumeID)
+}
+
+func (m *mounter) verifyVMwareDevice(devicePath string, volumeID string) bool {
```

Comment:
This method is same as verifyXenServerDevice. It will be better to use the same method here.